torch/csrc/jit/codegen/cuda/scheduler/matmul.cpp

```diff
@@ -0,0 +1,222 @@
+#include <torch/csrc/jit/codegen/cuda/scheduler/matmul.h>
+#include <torch/csrc/jit/codegen/cuda/scheduler/mma_utils.h>
+#include <torch/csrc/jit/codegen/cuda/scheduler/utils.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+
+namespace {
+// Move the broadcast axes to the left on the specified number of inner
+// dimensions e.g.  (when number_of_inner_pos == 3):
+//      [... I0, B, I1] -> [... B, I0, I1]
+//  should probably be only used to order innermost mnk axes.
+void moveInnerBroadcastLeft(TensorView* tv, int number_of_inner_pos = 3) {
+  TORCH_INTERNAL_ASSERT(tv->nDims() >= number_of_inner_pos);
+  std::vector<int> broadcast_pos;
+  std::vector<int> nonbroadcast_pos;
+
+  for (auto i : c10::irange(number_of_inner_pos)) {
+    auto axis_idx = i - number_of_inner_pos;
+    auto id = tv->axis(axis_idx);
+    if (id->isBroadcast()) {
+      broadcast_pos.push_back(axis_idx);
+    } else {
+      nonbroadcast_pos.push_back(axis_idx);
+    }
+  }
+
+  auto combined_pos_vec = broadcast_pos;
+  combined_pos_vec.insert(
+      combined_pos_vec.end(), nonbroadcast_pos.begin(), nonbroadcast_pos.end());
+
+  std::unordered_map<int, int> order_map;
+  for (auto i : c10::irange(number_of_inner_pos)) {
+    order_map[combined_pos_vec.at(i)] = i - number_of_inner_pos;
+  }
+
+  // Apply ordering.
+  tv->reorder(order_map);
+}
+
+} // namespace
+
+void scheduleMatmul(
+    TensorView* c,
+    TensorView* a,
+    TensorView* b,
+    MmaBuilder& mma_builder,
+    MatMulTileOptions& gemm_tile) {
+  // Currently only support a, b, c as fusion inputs/outputs
+  //  aka. no prolog and epilog fusion yet.
+  TORCH_CHECK(
+      c->isFusionOutput() && a->isFusionInput() && b->isFusionInput(),
+      "not supporting matmul fusion yet");
+  TORCH_CHECK(c->definition() && c->definition()->isA<MmaOp>());
+
+  mma_builder.configureMma(c);
```

Comment:
Thanks for reminding this. Yes I was planning on cleaning up the configuration steps gradually. 

`mma_builder.configureMma(c);` is still a necessary step but the `mma_op` doesn't need most of the info from`mma_builder`. The only thing that truly belong there is the actual mma macro type selected at scheduling phase and the rest of them should all be kept on the scheduled fusion graph or iterdomains. 

At current state, 2 parameters are kept on mma op still and would need another cleanup step to make it just the `macro_type`. 

I can do a minor cleanup in this PR to copy just the 2 needed parameters instead of dumping the whole `mma_options` class.

Reply:
Removed the un-necessary info from the mma node. Will remove more in follow up cleanups.